addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -59,12 +74,29 @@ public void init() throws IOException {
             dtoState = LEDStateDTO.valueOf(s.state, s.program, s.programSpeed, s.red, s.green, s.blue, s.white,
                     s.white2);
             power = (s.state & 0x01) != 0;
-            currentTargetState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
+            currentFaderState = currentState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
```

Comment:
Multiple assignments like this should better not be used, please make it more explicit:

```java
currentState = InternalLedState.fromRGBW(s.red, s.green, s.blue, s.white, s.white2);
currentFaderState = currentState;
```